src/suit_decoder.c

```diff
@@ -0,0 +1,822 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <suit_decoder.h>
+#include <suit_platform.h>
+#include <manifest_decode.h>
+#include <cose_encode.h>
+#include <cose_decode.h>
+#include <suit_manifest.h>
+
+/** Verify that the header matches the expected length.
+ */
+static bool header_check(const uint8_t *cbstr, size_t exp_len)
```

Comment:
A lot of of magic numbers.
I'm not saying they need to be defined, but at least add some reference document in the comment.
Same goes for header_len(...).

Reply:
This is pure CBOR and they are used only in this place. There is no need to go that deep down to the bytes in other places, so I do not see a value in making those defined.
This calculation should be handled by the ZCBOR, but unfortunately the function defined [here](https://github.com/NordicSemiconductor/zcbor/blob/cb82ae80b204170a89a49e2fc776fbdd220222b6/src/zcbor_common.c#L235C29-L235C29) is not robust enough.